src/Agent.Plugins/PipelineCache/PipelineCacheTaskPluginBase.cs

```diff
@@ -81,26 +85,35 @@ public async Task RunAsync(AgentTaskPluginExecutionContext context, Cancellation
             string salt = saltValue?.Value ?? string.Empty;
 
             VariableValue workspaceRootValue = context.Variables.GetValueOrDefault("pipeline.workspace");
-            string worksapceRoot = workspaceRootValue?.Value;
+            string workspaceRoot = workspaceRootValue?.Value;
 
             string key = context.GetInput(PipelineCacheTaskPluginConstants.Key, required: true);
             string restoreKeysBlock = context.GetInput(PipelineCacheTaskPluginConstants.RestoreKeys, required: false);
 
-            (bool isOldFormat, string[] keySegments, IEnumerable<string[]> restoreKeys) = ParseIntoSegments(salt, key, restoreKeysBlock);
+            (bool isOldFormat, IEnumerable<string> keySegments, IEnumerable<string[]> restoreKeys) = ParseIntoSegments(salt, key, restoreKeysBlock);
 
             if (isOldFormat)
             {
                 context.Warning(OldKeyFormatMessage);
             }
 
+            VariableValue packValue = context.Variables.GetValueOrDefault(PackingVariableName);
+            string pack = packValue?.Value ?? string.Empty;
+
+            if(String.IsNullOrWhiteSpace(pack))
```

Comment:
>IsNullOrWhiteSpace [](start = 22, length = 18)

Why does this get set if the variable is *not* set?

Reply:
That was just for testing, corrected this.